drivers/flash/flash_stm32wba_fm.c

```diff
@@ -66,17 +66,6 @@ bool flash_stm32_valid_range(const struct device *dev, off_t offset,
 	return flash_stm32_range_exists(dev, offset, len);
 }
 
-
-static inline void flash_stm32_sem_take(const struct device *dev)
-{
-	k_sem_take(&FLASH_STM32_PRIV(dev)->sem, K_FOREVER);
-}
-
-static inline void flash_stm32_sem_give(const struct device *dev)
-{
-	k_sem_give(&FLASH_STM32_PRIV(dev)->sem);
-}
-
```

Comment:
I see that these functions always operate on semaphore (no `#ifdef CONFIG_MULTITHREADING`). Should we change [this line](https://github.com/zephyrproject-rtos/zephyr/blob/6dafb5f4a9cbd7043989f986d1f5963c264d413e/drivers/flash/flash_stm32.h#L276C1-L276C72)?

Reply:
> Otherwise (when not required) we have an empty implementation.

Yes, but removed code doesn't provide no-op implementation if multithreading is disabled. That's why I asked about this. Since we also check `CONFIG_STM32H7_DUAL_CORE` in `flash_stm32.h` I suspect there are some cases where locking is desired even without multithreading enabled.

If there are no multicore `stm32wba` MCUs, then it should be fine, but please confirm.